workflow-manager/utils/utils.go

```diff
@@ -8,3 +22,62 @@ func Index(isFirst bool) int {
 	}
 	return 1
 }
+
+// Clock allows mocking of time for testing purposes
+type Clock struct {
+	now time.Time
+}
+
+// DefaultClock reutrns a Clock that returns the real time from time.Now
+func DefaultClock() Clock {
+	return Clock{}
+}
+
+// ClockWithFixedNow returns a Clock that always returns the provided now
+func ClockWithFixedNow(now time.Time) Clock {
+	return Clock{now: now}
+}
+
+// Now returns the current time according to this clock
+func (c *Clock) Now() time.Time {
+	if c.now.IsZero() {
+		return time.Now()
+	}
+	return c.now
+}
+
+func webIDP(sess *aws_session.Session, identity string) (*credentials.Credentials, error) {
+	parsed, err := arn.Parse(identity)
+	if err != nil {
+		return nil, err
+	}
+	audience := fmt.Sprintf("sts.amazonaws.com/%s", parsed.AccountID)
+
+	stsSTS := sts.New(sess)
+	roleSessionName := ""
+	roleProvider := stscreds.NewWebIdentityRoleProviderWithToken(
+		stsSTS, identity, roleSessionName, tokenfetcher.NewTokenFetcher(audience))
+
+	return credentials.NewCredentials(roleProvider), nil
+}
+
+// AWSClientConfig returns a (Session, Config) pair suitable for passing to the
+// New() functions for various AWS services. If identity contains a valid role
+// ARN, the config will use a web identity role provider for that role.
+func AWSClientConfig(region, identity string) (*aws_session.Session, *aws.Config, error) {
```

Comment:
I think it's best to have a very high bar for `util` packages, lest they become a crowded dumping ground. I think in this case there's a clear enough theme to put this function in its own `aws` package (where it would become `aws.ClientConfig`).

Reply:
Refactored this into the `ClientConfig` function in the `aws` package.